e2e/app/component/dialog.ts

```diff
@@ -1,57 +1,66 @@
+import Container from 'app/container';
+import {xPathOptionToXpath} from 'app/element/xpath-defaults';
 import {ElementHandle, Page} from 'puppeteer';
+import {ElementType} from '../xpath-options';
 
 export enum ButtonLabel {
   Confirm = 'Confirm',
   KeepEditing = 'Keep Editing',
+  Cancel = 'Cancel',
+  Calculate = 'Calculate',
+  AddThis = 'ADD THIS',
+  Finish = 'Finish',
 }
 
-const SELECTOR = {
-  dialogRoot: '.ReactModal__Content[role="dialog"]',
-  button: '[role="button"]'
+const Selector = {
+  defaultDialog: '//*[@role="dialog"]',
 }
 
-export default class Dialog {
-  private dialogElement: ElementHandle;
+export default class Dialog extends Container {
 
-  constructor(private readonly page: Page) {
+  private dialogElement: ElementHandle;
 
+  constructor(page: Page, xpath: string = Selector.defaultDialog) {
+    super(page, xpath);
   }
 
   async getContent(): Promise<string> {
     await this.findDialog();
-    const modalText = await this.page.evaluate((selector) => {
-      const modalElement = document.querySelector(selector);
-      return modalElement.innerText;
-    }, SELECTOR.dialogRoot);
-    console.log('dialog: ' + modalText);
-    return modalText;
+    const modalText = await (await this.dialogElement.getProperty('innerText')).jsonValue();
+    console.debug('dialog: \n' + modalText);
+    return modalText.toString();
   }
 
   async clickButton(buttonLabel: ButtonLabel): Promise<void> {
-    const selector = this.getButtonSelector();
-    await this.page.waitForSelector(selector, {visible: true});
-    const buttons = await this.page.$$(selector);
-    for (const button of buttons) {
-      const propValue = await button.getProperty('textContent');
-      if (await propValue.jsonValue() === buttonLabel) {
-        return await button.click();
-      }
-    }
-    throw new Error(`Failed to find button with label ${buttonLabel}`);
+    const button = await this.waitForButton(buttonLabel);
+    return button.click();
+  }
```

Comment:
extracted find button step out of `clickButton` function.

Reply:
might want to do this in one line. Would that be `await this.waitForButton(buttonLabel).then(b => b.click())`?